db/models.py

```diff
@@ -50,3 +54,74 @@
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="orders"
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+
+class Ticket(models.Model):
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    movie_session = models.ForeignKey(
+        to=MovieSession,
+        on_delete=models.CASCADE,
+        related_name="tickets"
+    )
+    order = models.ForeignKey(
+        to=Order,
+        on_delete=models.CASCADE,
+        related_name="tickets"
+    )
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=["row", "seat", "movie_session"],
+                name="ticket_constraint",
+            )
+        ]
+
+    def clean(self) -> None:
+        cinema_hall = self.movie_session.cinema_hall
+        if self.row > cinema_hall.rows:
+            raise ValidationError(
+                {
+                    "row": [
+                        "row number must be in available range: "
+                        f"(1, rows): (1, {cinema_hall.rows})"
+                    ]
+                }
+            )
+        if self.seat > cinema_hall.seats_in_row:
+            raise ValidationError(
+                {
+                    "seat": [
+                        "seat number must be in available range: "
+                        f"(1, seats_in_row): (1, {cinema_hall.seats_in_row})"
+                    ]
+                }
+            )
+
+    def save(self, *args, **kwargs) -> None:
+        self.full_clean()
```

Comment:
Calling `self.full_clean()` in the `save` method ensures that the `clean` method is executed before saving, which is a good practice to maintain data integrity.